src/components/GlobalPreferences/Notifications/NotificationsVerify.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Box, Button, GU, Info, LoadingRing, Text } from '@aragon/ui'
+import { verifyEmailToken } from './notification-service-api'
+import { VERIFY_SUBSECTION } from './constants'
+import checkmarkSvg from './check-mark.svg'
+
+export default function NotificationsVerify({
+  subsection,
+  onTokenChange,
+  onEmailChange,
+  navigateToNotifications,
+}) {
+  const [verified, setVerified] = useState(false)
+  const [error, setError] = useState(null)
+  useEffect(() => {
+    // Parse token from subsection /verify/TOKEN -> TOKEN
+    const token = subsection.substring(VERIFY_SUBSECTION.length)
+    verifyEmailToken(token)
+      .then(longLivedToken => {
+        onTokenChange(longLivedToken)
+        setVerified(true)
+        return longLivedToken
+      })
+      .catch(e => {
+        // if an invalid token is passed. Clear email and token to reset state
+        setVerified(false)
+        setError(e.message)
+        onEmailChange(null)
+        onTokenChange(null)
+      })
+  }, [subsection, onTokenChange, onEmailChange])
+
+  if (!verified && !error) {
+    return (
+      <Box heading="Email notifications">
+        <LoadingRing />
```

Comment:
How about wrapping `LoadingRing` in a container and centering it: We should also give the container the same height as the next elements being rendered ("Verification successful") to prevent a jump in height

Reply:
Makes sense with the height.

As for centring, you mean horizontally right?

Reply:
and vertically...